### PR TITLE
Fixing ArgumentError: wrong number of arguments (given 0, expected 1) in Rails 4.2.7

### PR DESCRIPTION
**Pull request by sullerandras**

### CODE REVIEW

**Comment by jmgarnier** [comment]
I also have the same problem when using `config.logger = Logger.new(STDOUT)`

@stve Do you need more info / code to merge this PR? May I help?

File: spec/silencer/rails/logger_spec.rb [hunk 1]
```diff
@@ -78,4 +79,14 @@
...
     logger.call(Rack::MockRequest.env_for('/'))
   end
+
+  context 'when logger is Syslogger' do
+    it 'does not throw error' do
+      mock_rails_logger = Syslogger.new
+      allow(::Rails).to receive(:logger) { mock_rails_logger }
+
+      logger = Silencer::Rails::Logger.new(app, silence: ['/'])
+      expect { logger.call(Rack::MockRequest.env_for('/')) }.to_not raise_error
```
Comment (stve) [inline spec/silencer/rails/logger_spec.rb]: Would you mind changing this to `not_to`. There's nothing wrong with `to_not` just a personal preference 😄

File: lib/silencer/rails/logger.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@ def call(env)
       private
 
       def quiet(&block)
-        if ::Rails.logger.respond_to?(:silence)
+        if ::Rails.logger.respond_to?(:silence) && ::Rails.logger.method(:silence).owner != ::Kernel
```
Comment (stve) [inline lib/silencer/rails/logger.rb]: Can you explain the rationale around `method(:silence).owner != ::Kernel`. 

I've been under the impression that `Rails.logger.silence` was a Rails-only implementation, what does the `owner != ::Kernel` safeguard against?

Comment (sullerandras) [inline lib/silencer/rails/logger.rb]: In Rails 4 they defined `silence` multiple times, one here: [Kernel#silence](https://github.com/rails/rails/blob/v4.2.7/activesupport/lib/active_support/core_ext/kernel/reporting.rb#L108) (with a mandatory parameter, we don't want to call this), the other one is [LoggerSilence#silence](https://github.com/rails/rails/blob/v4.2.7/activesupport/lib/active_support/logger_silence.rb#L13) (with an optional parameter, we want to call this).

I got the guarding code from this commit: https://github.com/rails/rails/commit/33015969d1e585073709f7803515acf19e3b2d5f

File: spec/silencer/rails/logger_spec.rb (hunk 1 above)
Comment (sullerandras) [inline spec/silencer/rails/logger_spec.rb]: @stve Updated the PR if you want to have another look.

File: lib/silencer/rails/logger.rb (hunk 2 above)
Comment (stve) [inline lib/silencer/rails/logger.rb]: Thanks for the explanation. 🙇

**Comment by stve** [comment]
Thanks for turning those around so quickly @sullerandras!

I've just released `1.0.1` with your changes 🎉

**Pull request merged**